scripts/code_style.py

```diff
@@ -75,8 +75,24 @@ def get_src_files(since: Optional[str]) -> List[str]:
     output = subprocess.check_output(["git", "ls-files"] + file_patterns,
                                      universal_newlines=True)
     src_files = output.split()
+
+    # When this script is called from a git hook, some environment variables
+    # are set by default which force all git commands to use the main repository
+    # (i.e. prevent us from performing commands on the framework repo).
+    # Create an environment without these variables for running commands on the
+    # framework repo.
+    framework_env = os.environ.copy()
+    # Get a list of environment vars that git sets
+    git_env_vars = subprocess.check_output(["git", "rev-parse", "--local-env-vars"],
+                                           universal_newlines=True)
+    # Remove the vars from the environment
+    for var in git_env_vars.split():
+        framework_env.pop(var, None)
+
     output = subprocess.check_output(["git", "-C", "framework", "ls-files"]
```

Comment:
I think the solution here is actually on the wrong track, although it's an improvement so I won't block it. When this script is called with `--since`, it looks for the given revision name in the submodule. Reported as https://github.com/Mbed-TLS/mbedtls-framework/issues/24. I suspect the right solution would be to never call `git -C framework` and instead rely on `--recurse-submodules` throughout.

Reply:
Using `--recurse-submodules` comes with some issues as well: see https://github.com/Mbed-TLS/mbedtls/pull/9088/commits/7661aa0e204eed21e6408bd1a6acdd9a1c571535 commit message.